src/main/java/address/model/datatypes/ViewablePerson.java

```diff
@@ -0,0 +1,214 @@
+package address.model.datatypes;
+
+import javafx.beans.property.*;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+/**
+ * Extends {@link Viewable} for the Person domain object.
+ * @see Viewable
+ */
+public class ViewablePerson extends Viewable<Person> implements ObservableViewablePerson {
+
+    private final BooleanProperty isDeleted;
+    private final BooleanProperty isEdited;
+
+    {
+        isDeleted = new SimpleBooleanProperty(false);
+        isEdited = new SimpleBooleanProperty(false);
+    }
+
+    public ViewablePerson(Person backingPerson) {
+        super(backingPerson, Person::new);
+    }
+
+    @Override
+    protected void conditionallyBindVisibleToBacking() {
+        forceSyncFromBacking();
+        backing.forEachPropertyFieldPairWith(visible, this::conditionallyBindValue);
+
+        // all changes here result in removal of affected list elements and (re)adding of any updated/new elements
+        // at their correct positions.
+        backing.getTags().addListener((ListChangeListener<? super Tag>) change -> {
+            while (change.next()) {
+                int from = change.getFrom();
+                int to = change.getTo();
+                if (change.wasPermutated()) { // element reordering
+                    visible.getTags().subList(from, to).clear();
+                    visible.getTags().addAll(from, backing.getTags().subList(from, to));
+                } else { // list/element mutation
+                    visible.getTags().subList(from, from + change.getRemovedSize()).clear();
+                    visible.getTags().addAll(from, backing.getTags().subList(from, from + change.getAddedSize()));
+                }
+            }
+        });
+    }
+
+    @Override
+    public void forceSyncFromBacking() {
+        visible.update(backing);
+    }
+
+// APPLICATION STATE ACCESSORS
+
+    @Override
+    public boolean isDeleted() {
+        return isDeleted.get();
+    }
+
+    @Override
+    public BooleanProperty isDeletedProperty() {
+        return isDeleted;
+    }
+
+    public void setIsDeleted(boolean isDeleted) {
+        this.isDeleted.set(isDeleted);
+    }
+
+    @Override
+    public boolean isEdited() {
+        return isEdited.get();
+    }
+
+    @Override
+    public BooleanProperty isEditedProperty() {
+        return isEdited;
+    }
+
+    public void setIsEdited(boolean isEdited) {
+        this.isEdited.set(isEdited);
+    }
+
+
```

Comment:
Will add comments explaining the state properties after I finalise the "visible state" implementation in the `EditInProgress` PR